src/vs/workbench/services/editor/common/filteredEditorGroup.ts

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
As discussed I would suggest to change this to a class that purely works on the `EditorGroupModel` from `editorGroupModel.ts` and not on the group viewer.

Here is a sketch:
- extract a `IReadonlyEditorGroupModel` interface from `EditorGroupModel` and let `EditorGroupModel` implement it
- have a `FilteredReadonlyEditorGroupModel` and related subclasses for sticky/unsticky as you have done here
- have this in a file next to `editorGroupModel.ts` or inside that file, I don't mind
- move related tests over to `src/vs/workbench/test/browser/parts/editor` where the other editor group model tests are
- use `IReadonlyEditorGroupModel` in tab controls as you do today